modules/core/src/main/java/org/apache/ignite/internal/processors/cache/ClearCachesTask.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2024 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.cache;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.ignite.Ignite;
+import org.apache.ignite.IgniteCache;
+import org.apache.ignite.IgniteException;
+import org.apache.ignite.internal.processors.task.GridInternal;
+import org.apache.ignite.internal.visor.VisorJob;
+import org.apache.ignite.internal.visor.VisorOneNodeTask;
+import org.apache.ignite.resources.IgniteInstanceResource;
+import org.jetbrains.annotations.Nullable;
+
+/** Clears specified caches. */
+@GridInternal
+public class ClearCachesTask extends VisorOneNodeTask<ClearCachesTaskArg, ClearCachesTaskResult> {
+    /** */
+    private static final long serialVersionUID = 0L;
+
+    /** */
+    @Override protected VisorJob<ClearCachesTaskArg, ClearCachesTaskResult> job(ClearCachesTaskArg arg) {
+        return new ClearCacheJob(arg, debug);
+    }
+
+    /** Job clears specified caches. */
+    private static class ClearCacheJob extends VisorJob<ClearCachesTaskArg, ClearCachesTaskResult> {
+        /** */
+        private static final long serialVersionUID = 0L;
+
+        /** Local Ignite instance. */
+        private Ignite ignite;
+
+        /** */
+        private ClearCacheJob(ClearCachesTaskArg arg, boolean debug) {
+            super(arg, debug);
+        }
+
+        /** {@inheritDoc} */
+        @Override protected ClearCachesTaskResult run(@Nullable ClearCachesTaskArg arg) throws IgniteException {
+            List<String> clearedCaches = new ArrayList<>();
+            List<String> nonExistentCaches = new ArrayList<>();
+
+            for (String cache: arg.caches()) {
+                IgniteCache<?, ?> ignCache = ignite.cache(cache);
+
+                if (ignCache == null)
+                    nonExistentCaches.add(cache);
+                else {
+                    ignCache.clear();
```

Comment:
This operation may take a long time.
Perhaps, it makes sense to use https://www.gridgain.com/sdk/latest/javadoc/org/apache/ignite/compute/ComputeJobContinuation.html

Reply:
With the change to the compute job, creating the list of modified caches is more challenging. The AI version outputs which caches were cleared in the console after the command runs. The new GG version does not.

AI Output
```
Time: 2024-07-28T18:42:09.966
Command [CACHE] started
Arguments: --cache clear --caches SQL_PUBLIC_PERSON,sam 
--------------------------------------------------------------------------------
The following caches have been cleared: SQL_PUBLIC_PERSON
The following caches don't exist: sam
Command [CACHE] finished with code: 0
Control utility has completed execution at: 2024-07-28T18:42:11.912
Execution time: 1946 ms
```

GG Output
```
Command [CACHE] started
Arguments: --cache clear --caches SQL_PUBLIC_PERSON,sam
--------------------------------------------------------------------------------
Command [CACHE] finished with code: 0
Control utility has completed execution at: 2024-07-28T20:41:49.660
Execution time: 2743 ms
```

Do you think it is necessary to output this information?
if yes: do you have a recommendation? I have not had success with building the lists inside the job
if no: do you think it makes sense to clean up the output lines in [modules/core/src/main/java/org/apache/ignite/internal/visor/cache/ClearCachesTaskArg.java](https://github.com/gridgain/gridgain/pull/3139/files#diff-d42cf87402346a759eb2061f6a5d0fbd79020810dcb163e60937c50379770142)
and [modules/core/src/main/java/org/apache/ignite/internal/visor/cache/ClearCachesTaskResult.java](https://github.com/gridgain/gridgain/pull/3139/files#diff-0d4a5235a4bef482476dae9f3cf5134efb5ba43ee1b32e5c1f69ffaea6fda382)?

Reply:
I think it can be done in the following way:
```
    /** Job clears specified caches. */
    private static class ClearCacheJob extends VisorJob<ClearCachesTaskArg, ClearCachesTaskResult> {
        ...
        private GridCompoundFuture opFut;

        List<String> clearedCaches = new ArrayList<>();
        List<String> nonExistentCaches = new ArrayList<>();
        ...

        /** {@inheritDoc} */
        @Override protected ClearCachesTaskResult run(@Nullable ClearCachesTaskArg arg) throws IgniteException {
            if (opFut == null) {
                opFut = new GridCompoundFuture<>();

                for (String cache: arg.caches()) {
                    IgniteCache<?, ?> ignCache = ignite.cache(cache);

                    if (ignCache == null) {
                        nonExistentCaches.add(cache);
                        continue;
                    }
                    else {
                        GridFutureAdapter<?> fut = new GridFutureAdapter<>();

                        ignCache.clearAsync().listen(asyncClearFut -> {
                            try {
                                asyncClearFut.get();
                            }
                            catch (IgniteException e) {
                                fut.onDone(e);

                                return;
                            }

                            clearedCaches.add(cache);

                            fut.onDone();
                        });

                        opFut.add(fut);
                    }
                }

                jobCtx.holdcc();

                opFut.listen(f -> jobCtx.callcc());

                opFut.markInitialized();

                return null;
            }

            assert opFut.isDone();

            return new ClearCachesTaskResult(clearedCaches, nonExistentCaches);
        }
    }

```

Reply:
I appreciate your assistance. I see where I was going wrong.
Thank you.